### ISSUE TITLE
Fix error messages for the JSON commands

### ISSUE THREAD

**Issue by BagritsevichStepan** [dragonflydb/dragonfly#3777]
Sometimes users receive other errors unlike Redis for JSON. In addition, error messages may vary depending on the version of the json path (V2 or legacy).

The errors are also different for `--jsonpathv2=true` and `--jsonpathv2=false`. This is unexpected behavior and should be fixed.

**Comment by romange** [comment]
True, but I consider these issues as really minor, because I am not aware of specific user usecases that break because of it.
